src/solace_ai_connector/components/inputs_outputs/websocket_base.py

```diff
@@ -0,0 +1,143 @@
+"""Base class for WebSocket components."""
+
+from abc import ABC, abstractmethod
+from flask import Flask, send_file, request
+from flask_socketio import SocketIO
+import logging
+from ...common.log import log
+from ..component_base import ComponentBase
+import copy
+from flask.logging import default_handler
+
+base_info = {
+    "config_parameters": [
+        {
+            "name": "listen_port",
+            "type": "int",
+            "required": False,
+            "description": "Port to listen on (optional)",
+        },
+        {
+            "name": "serve_html",
+            "type": "bool",
+            "required": False,
+            "description": "Serve the example HTML file",
+            "default": False,
+        },
+        {
+            "name": "html_path",
+            "type": "string",
+            "required": False,
+            "description": "Path to the HTML file to serve",
+            "default": "examples/websocket/websocket_example_app.html",
+        },
+        {
+            "name": "payload_encoding",
+            "required": False,
+            "description": "Encoding for the payload (utf-8, base64, gzip, none)",
+            "default": "none",
+        },
+        {
+            "name": "payload_format",
+            "required": False,
+            "description": "Format for the payload (json, yaml, text)",
+            "default": "json",
+        },
+    ],
+}
+
+
+class WebsocketBase(ComponentBase, ABC):
+    def __init__(self, info, **kwargs):
+        super().__init__(info, **kwargs)
+        self.listen_port = self.get_config("listen_port")
+        self.serve_html = self.get_config("serve_html", False)
+        self.html_path = self.get_config("html_path", "")
+        self.sockets = {}
+        self.app = None
+        self.socketio = None
+
+        if self.listen_port:
+            self.setup_websocket_server()
+
+    def setup_websocket_server(self):
+        self.app = Flask(__name__)
+
+        # Enable Flask debugging
+        self.app.debug = False
+
+        # Set up Flask logging
+        # self.app.logger.setLevel(logging.DEBUG)
+        # self.app.logger.addHandler(default_handler)
+
+        # Enable SocketIO logging
+        # logging.getLogger("socketio").setLevel(logging.DEBUG)
+        # logging.getLogger("engineio").setLevel(logging.DEBUG)
+
+        self.socketio = SocketIO(
+            self.app, cors_allowed_origins="*", logger=False, engineio_logger=False
+        )
+        self.setup_websocket()
+
+        if self.serve_html:
+            self.setup_html_route()
+
+    def setup_html_route(self):
+        @self.app.route("/")
+        def serve_html():
+            return send_file(self.html_path)
```

Comment:
Wouldn't serve directory be bettor? to support css, js, and asset files?

Reply:
I considered this and I could be wrong with my choice. I didn't think it really made sense to use this as a general webserver, but I wanted a really simple demo that wouldn't require you to create a webserver... So I just served a single file. By doing this, you don't need to know the name of the file being served.